src/page_parser.py

```diff
@@ -238,23 +240,26 @@ def data_push_fontname(self, fontdata: dict):
                     self.all_fontnames[name] += num
 
     def add_bbox(
-        self, bbox: tuple, color: tuple | None = None, fillColor: tuple | None = None
+        self, bbox: tuple, color: tuple or None = None, fillColor: tuple or None = None
     ):
         self.bboxes.append(bbox)
         self.fill_bbox(bbox, color, fillColor)
 
     def add_bbox_content(
-        self, text: str | None, color: tuple | None = None, fontdata: dict | None = None
+        self,
+        text: str or None,
+        color: tuple or None = None,
```

Comment:
I know your just refactoring the code and you didn't write the original code, but do you know why `color` is type to be tuple or `None=None`. The part I don't understand is why is it `None = None` instead of it just being `None` like the variable type for `text`

Reply:
It looks like this is a method where you can choose to add a color to the bbox, but if you don't it auto inits to none. So parameter color is type tuple or None, and if you don't specify a parameter it defaults to value None.

That's what I'm guessing. The types don't have to be specified, I feel like that is part of the confusion here.

@chp2001 I'm assuming this is your code bc you did the box stuff. Can you confirm, I could be wrong?